miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -645,15 +729,45 @@ end
 #! Returns the serial number of the note currently being processed.
 #! Panics if no note is not being processed.
 #!
-#! Inputs: []
+#! Inputs: [KERNEL_PROCEDURE_HASH]
 #! Outputs: [SERIAL_NUMBER]
 #!
 #! - SERIAL_NUMBER is the serial number of the note currently being processed.
 export.get_note_serial_number
+    # drop the procedure's hash
+    dropw
+    
     exec.note::get_serial_number
     # => [SERIAL_NUMBER]
 
     # drop the padding
     swapw dropw
     # => [SERIAL_NUMBER]
 end
+
+#! Executes a kernel procedure specified by its offset.
+#!
+#! Inputs: [procedure_offset, *executed_procedure_input*, PAD(*)]
+#! Outputs: [*executed_procedure_output*, PAD(*)]
```

Comment:
nit: `executed_procedure_input` and `executed_procedure_output` sounds a bit odd. I would probably use `procedure_inputs` and `procedure_outputs`.

Also, we should explain what these are in the comments. For example, we should make it clear that the number of procedure inputs cannot exceed 12 elements.